### PR TITLE
Adds track rotation and auto-save/ load to foulplay

### PR DESCRIPTION
**Pull request by notjustmat**
Had a bit of a git hiccup, but everything is good now. 

updates foulplay.lua with rotation per track, and save/ recall. also updates foulplay.md with the new details.

### CODE REVIEW

File: docs/justmat/foulplay/foulplay.md [hunk 1]
```diff
@@ -65,6 +61,15 @@ copy to return to normal use.
 
 # extended explanations
 
+### saving/ loading your work
+
+foulplay will automatically save its _current state*_ whenever (a) you switch scripts, or (b) when norns _sleeps_.
```
Comment (pq) [inline docs/justmat/foulplay/foulplay.md]: this is a footnote? aside: i wonder if there's a better way...  i can't recall how these markdown files are getting processed but it'd be nice to figure out something a bit more elegant (and capture it in wiki/FAQ).

File: scripts/justmat/foulplay.lua [hunk 2]
```diff
@@ -86,8 +80,10 @@ local BeatClock = require 'beatclock'
 
 local clk = BeatClock.new()
 
+local ALT = 0
```
Comment (pq) [inline scripts/justmat/foulplay.lua]: no biggie but curious why the screaming caps?

File: docs/justmat/foulplay/foulplay.md (hunk 1 above)
Comment (notjustmat) [reply docs/justmat/foulplay/foulplay.md]: I'm sure there is a better way. I wanted to include the saving feature in the docs, but as there is no user interaction in saving... I was unsure how to go about it. Should I just omit this part?

File: scripts/justmat/foulplay.lua (hunk 2 above)
Comment (notjustmat) [reply scripts/justmat/foulplay.lua]: Haha, no reason at all. I will turn down the volume now :)

**Comment by notjustmat** [comment]
alt is much more user friendly now :D

**Pull request merged**